examples/simple.rs

```diff
@@ -425,7 +425,7 @@ impl SimpleFS {
 }
 
 impl Filesystem for SimpleFS {
-    fn init(&mut self, _req: &Request, _config: &mut KernelConfig) -> Result<(), c_int> {
+    fn init(&self, _req: &Request, _config: &mut KernelConfig) -> Result<(), c_int> {
```

Comment:
for something to be async friendly requiring interior mutability in the file system seems to be really a requirement